### ISSUE TITLE
[BUG/Feature Request] adding signed PDF documents

### ISSUE THREAD

**Issue by GarryG** [paperless-ngx/paperless-ngx#4987]
### Description

I just received some documents that are signed by some issuer. When I try adding them to paperless, I get this error message:

DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document, invalidating the signature.

All I could find in the closed issues were cases where users reported that the signatures were broken upon upload due to OCR information being placed inside the PDF. If the current behavior is a result of those tickets, it might be nice if paperless would store both the original file (for exact retrieval) as well as the OCR'ed version, allowing for the user to download either version. Alternatively, allow for storing the original document without alterations, skipping OCR.

### Steps to reproduce

1. add a signed document (see sample doc generated by docusign.com:
[Here_is_your_signed_document_test_unsignedp.pdf](https://github.com/paperless-ngx/paperless-ngx/files/13684120/Here_is_your_signed_document_test_unsignedp.pdf) )
2. admire the error message

### Webserver logs

```bash
Considering the error message, most likely not an error but deliberate non-processing
```

### Paperless-ngx version

2.1.2

### Host OS

Docker images on Ubuntu Srv

### Installation method

Docker - official image

**Comment by shamoon** [comment]
As you noted there are many existing issues for this. A recent change in ocrmypdf did change how things handled (by that tool) but paperless provides the ability to set the options to allow this.

if you want to open a FR they go under “Discussions”. (There might be one for this already, I don’t recall)

**Comment by suhrmann** [comment]
### What happens
When I uploaded a signed PDF the PDF was not added to Paperless, but stopped with this error (same in logs):
> DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document, invalidating the signature. 

### Fix
Setting this config value fixed the issue for me (Paperless-ngx 2.1.3):
```
PAPERLESS_OCR_SKIP_ARCHIVE_FILE = with_text
```
> [`PAPERLESS_OCR_SKIP_ARCHIVE_FILE=<mode>`](https://docs.paperless-ngx.com/configuration/#PAPERLESS_OCR_SKIP_ARCHIVE_FILE)
> Specify when you would like paperless to skip creating an archived version of your documents. 
> The default is never.
> `with_text`: Skip creating an archived version for documents that already have embedded text.
> [...]

-----

### Unexpected behaviour (imho)
Nevertheless, I would expect from the default value to always add the PDF to Paperless, aka always (re-)create the archived PDF - and not stopping with the mentioned DigitalSignatureError:
> [PAPERLESS_OCR_SKIP_ARCHIVE_FILE=<mode>](https://docs.paperless-ngx.com/configuration/#PAPERLESS_OCR_SKIP_ARCHIVE_FILE)
> `never`: Never skip creating an archived version.
> [...] The default is `never`.
